src/Pdo/PdoInsertCapableTrait.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace RebelCode\Storage\Resource\Pdo;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use PDOStatement;
+
+/**
+ * Common functionality for objects that can insert records into a database using PDO.
+ *
+ * @since [*next-version*]
+ */
+trait PdoInsertCapableTrait
+{
+    /**
+     * Executes an INSERT SQL query, inserting records into the database.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $records A list of containers, each containing the record data to insert.
+     *
+     * @return PDOStatement The executed PDO statement.
+     */
+    protected function _insert(array $records)
+    {
+        $valueHashMap = array_map([$this, '_getSqlInsertValueHashMap'], $records);
+
+        $query = $this->_buildInsertSql(
+            $this->_getSqlInsertTable(),
+            $this->_getSqlInsertColumnNames(),
+            $records,
+            $valueHashMap
+        );
+
+        $statement = $this->_executePdoQuery($query);
+
+        return $statement;
+    }
+
+    /**
+     * Extracts and retrieves the data hash map from record data.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $record An associative array mapping field names to their values for a single record.
+     *
+     * @return array The generated value hash map, mapping values to their respective hashes.
```

Comment:
How about I create a new `MapInterface extends ContainerInterface, Traversable`, and you can return that?

Reply:
I don't think this is an adequate place to be dealing with high-level data structures. First off, this is not API-level, but implementation logic code, that will feed its result into another non-API method. Secondly, it would require a factory - which is added time and effort that I don't think is merited.